File: user-management-service/src/main/java/com/platform/usermanagement/event/UserCreatedEvent.java
package com.platform.usermanagement.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Event published when a new user is created.
 * Published to Kafka for analytics and downstream processing.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserCreatedEvent {

    private UUID userId;
    private String email;
    private String name;
    private LocalDateTime createdAt;
    private String eventType;

    public static UserCreatedEvent from(UUID userId, String email, String name, LocalDateTime createdAt) {
        return UserCreatedEvent.builder()
                .userId(userId)
                .email(email)
                .name(name)
                .createdAt(createdAt)
                .eventType("USER_CREATED")
                .build();
    }
}

File: src/UserManagementService/Models/Events/UserCreatedEvent.cs
namespace UserManagementService.Models.Events;

/// <summary>
/// Event published when a new user is created.
/// TODO: Confirm Kafka event publishing requirements with domain experts.
/// </summary>
public class UserCreatedEvent
{
    public Guid UserId { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public string EventType { get; set; } = "UserCreated";
    public DateTime EventTimestamp { get; set; } = DateTime.UtcNow;
}
